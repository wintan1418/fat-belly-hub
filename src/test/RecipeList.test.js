import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import RecipeList from '../container/Recipe';

describe('Creates a component', () => {
  const middlewares = [thunk];
  const initialState = { meals: ['goat'] };
  const mockStore = configureStore(middlewares);
  let store;
  it('returns correctly', () => {
    store = mockStore(initialState);
    const tree = renderer.create(
      <Provider store={store}>
        <RecipeList />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
