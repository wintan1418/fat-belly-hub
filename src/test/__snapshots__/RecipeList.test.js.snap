// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a component returns correctly 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": 40,
    }
  }
  textalign="center"
>
  <div
    className="ui three column grid"
  >
    <div
      className="row"
    />
  </div>
</div>
`;
